lop.PartitionBy does not guarantee the same order

I have tested the example of[ lop.PartitionBy on README.md](https://github.com/samber/lo?tab=readme-ov-file#partitionby)
It keep changed the order

sorry for confusion. I mean the parallel version. belows are my test and the result
```
func ExamplePartitionByParallel3() {
	partitions := lop.PartitionBy([]int{-2, -1, 0, 1, 2, 3, 4, 5}, func(x int) string {
		if x < 0 {
			return "negative"
		} else if x%2 == 0 {
			return "even"
		}
		return "odd"
	})
	fmt.Println(partitions)
	// Output:
	// [[-2 -1], [0 2 4], [1 3 5]]
}
```

result
```
=== RUN   ExamplePartitionByParallel3
--- FAIL: ExamplePartitionByParallel3 (0.00s)
got:
[[5 1 3] [2 4 0] [-2 -1]]
want:
[[-2 -1], [0 2 4], [1 3 5]]
```

--- a/parallel/slice.go
+++ b/parallel/slice.go
@@ -66,28 +66,18 @@ func Times[T any](count int, iteratee func(index int) T) []T {
 }
 
 // GroupBy returns an object composed of keys generated from the results of running each element of collection through iteratee.
+// The order of grouped values is determined by the order they occur in the collection.
 // `iteratee` is call in parallel.
 func GroupBy[T any, U comparable, Slice ~[]T](collection Slice, iteratee func(item T) U) map[U]Slice {
 	result := map[U]Slice{}
 
-	var mu sync.Mutex
-	var wg sync.WaitGroup
-	wg.Add(len(collection))
+	keys := Map(collection, func(item T, _ int) U {
+		return iteratee(item)
+	})
 
-	for _, item := range collection {
-		go func(_item T) {
-			key := iteratee(_item)
-
-			mu.Lock()
-
-			result[key] = append(result[key], _item)
-
-			mu.Unlock()
-			wg.Done()
-		}(item)
+	for i, item := range collection {
+		result[keys[i]] = append(result[keys[i]], item)
 	}
-
-	wg.Wait()
 
 	return result
 }
@@ -95,36 +85,25 @@ func GroupBy[T any, U comparable, Slice ~[]T](collection Slice, iteratee func(it
 // PartitionBy returns an array of elements split into groups. The order of grouped values is
 // determined by the order they occur in collection. The grouping is generated from the results
 // of running each element of collection through iteratee.
+// The order of groups is determined by their first appearance in the collection.
 // `iteratee` is call in parallel.
 func PartitionBy[T any, K comparable, Slice ~[]T](collection Slice, iteratee func(item T) K) []Slice {
 	result := []Slice{}
 	seen := map[K]int{}
 
-	var mu sync.Mutex
-	var wg sync.WaitGroup
-	wg.Add(len(collection))
+	keys := Map(collection, func(item T, _ int) K {
+		return iteratee(item)
+	})
 
-	for _, item := range collection {
-		go func(_item T) {
-			key := iteratee(_item)
-
-			mu.Lock()
-
-			resultIndex, ok := seen[key]
-			if !ok {
-				resultIndex = len(result)
-				seen[key] = resultIndex
-				result = append(result, []T{})
-			}
-
-			result[resultIndex] = append(result[resultIndex], _item)
-
-			mu.Unlock()
-			wg.Done()
-		}(item)
+	for i, item := range collection {
+		if resultIndex, ok := seen[keys[i]]; ok {
+			result[resultIndex] = append(result[resultIndex], item)
+		} else {
+			resultIndex = len(result)
+			seen[keys[i]] = resultIndex
+			result = append(result, Slice{item})
+		}
 	}
-
-	wg.Wait()
 
 	return result
 }

